runtime/deployedcontract_test.go

```diff
@@ -0,0 +1,114 @@
+package runtime
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/cadence"
+	"github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/cadence/runtime/tests/utils"
+)
+
+func TestDeployedContracts(t *testing.T) {
+	t.Parallel()
+
+	contractCode := `
+		pub contract Test {
+			pub struct A {}
+			pub struct B {}
+			pub struct C {}
+
+			init() {}
+		}
+	`
+
+	script :=
+		`
+		transaction {
+			prepare(signer: AuthAccount) {
+				let deployedContract = signer.contracts.get(name: "Test")
+				assert(deployedContract!.name == "Test")
+
+				let expected: {String: Void} =  
+					{ "A.2a00000000000000.Test.A": ()
+					, "A.2a00000000000000.Test.B": ()
+					, "A.2a00000000000000.Test.C": ()
+					}
+				let types = deployedContract!.publicTypes()
+				assert(types.length == 3)
+
+				for type in types {
+					assert(expected[type.identifier] != nil, message: type.identifier)
```

Comment:
maybe make this a script, return the types from the script, and assert them at Go level.  It gives more control over the assertion: e.g: Can check for the types-kind, names, etc.

Reply:
ah yeah, that was my original idea but faced some roadblocks when trying to access the contract itself from the script. 

when we deploy the contracts to the runtime manually using `rt.ExecuteTransaction`, are they stored under the signer's address? what address should we be passing to `getAccount` inside Cadence?

Reply:
> when we deploy the contracts to the runtime manually using rt.ExecuteTransaction, are they stored under the signer's address? what address should we be passing to `getAccount` inside Cadence?

Yes. Should be able to get the account using the same address.

Alternatively, can always return the same contract code inside `GetAccountContractCode` without really worrying about the address, since there's only one contract anyway.